### PR TITLE
docs: Add upgrading page

### PR DESCRIPTION
**Pull request by markgoddard**
- docs: Add upgrading page

### CODE REVIEW

**Comment by markgoddard** [comment]
Currently just borrowed upstream Kayobe upgrade page, needs more info.

File: doc/source/operations/upgrading.rst [hunk 1]
```diff
@@ -0,0 +1,726 @@
+=========
+Upgrading
+=========
+
+This section describes how to upgrade from the previous OpenStack release
+series to |current_release|. It is based on the :kayobe-doc:`upstream Kayobe
+documentation <upgrading>` with additional considerations for using StackHPC
+Kayobe Configuration.
+
+Notable changes in the |current_release| Release
```
Comment (Alex-Welsh) [inline doc/source/operations/upgrading.rst:32]: One thing I found when updating SKC for Zed was that it was nice and easy to just grep through the repo for "yoga" to find instances of things to edit. Making things like this a variable is very useful when they don't change between releases, but this page in particular will need a rewrite for each release. I think just plaintext writing `zed` here would make more sense.

File: doc/source/operations/upgrading.rst [hunk 2]
```diff
@@ -0,0 +1,726 @@
...
+    git checkout -b example/|current_release|-sync origin/example/|current_release|
+    git merge stackhpc/|current_release_git_branch_name|
+
+There may be conflicts to resolve. The configuration should be manually
+inspected after the merge to ensure that it is correct. Once complete, push the
+branch and create a pull request with the changes::
+
+    git push origin example/|current_release|-sync
+
+Once approved and merged, update the configuration to adapt to the new release.
```
Comment (Alex-Welsh) [inline doc/source/operations/upgrading.rst:200]: Might be worth reminding people here that the base environment ``etc/kayobe/foo.yml`` will likely have changes that are brought in from the merge, but anything in a deployment-specific environment will be completely unaffected.

File: doc/source/operations/upgrading.rst (hunk 1 above)
Comment (markgoddard) [reply doc/source/operations/upgrading.rst:32]: Would it help to have a special release-agnostic comment that you could search for? It seems a shame to hard code things that don't generally need to change.

```
# NOTE(upgrade): Update these notable changes for the current release.
```

File: doc/source/operations/upgrading.rst (hunk 2 above)
Comment (markgoddard) [reply doc/source/operations/upgrading.rst:200]: Good idea.

File: doc/source/operations/upgrading.rst (hunk 1 above)
Comment (Alex-Welsh) [reply doc/source/operations/upgrading.rst:32]: Sounds good, I'll start adding this in Antelope too.

File: doc/source/configuration/wazuh.rst [hunk 3]
```diff
@@ -302,6 +302,8 @@ Encrypt the keys (and remember to commit to git):
 
 ``ansible-vault encrypt --vault-password-file ~/vault.pass $KAYOBE_CONFIG_PATH/ansible/wazuh/certificates/certs/*.key``
 
+.. _wazuh-verification:
```
Comment (Alex-Welsh) [inline doc/source/configuration/wazuh.rst:305]: Where's this coming from?

File: doc/source/operations/upgrading.rst [hunk 4]
```diff
@@ -0,0 +1,919 @@
...
+Pulling container images from the local Pulp server to the control plane hosts
+can take a considerable time, because images are only synced from Ark to the
+local Pulp on demand, and there is potentially a large fan-out. Pulling images
+in advance of the upgrade moves this step out of the maintenance/change window.
+
+.. code-block:: console
+
+   kayobe overcloud container image pull
+
+Preview overcloud service configuration changes
```
Comment (Alex-Welsh) [inline doc/source/operations/upgrading.rst:371]: I tried this config diff process during the Zed upgrade on SMS and something went a bit wrong. I'm not sure if it was an issue with my config or the commands & I don't remember exactly what i used to correct it. I'll try it again for Antelope and make a separate PR with changes if needed. I'm happy with this as-is for now, just wanted to note my experience

File: doc/source/operations/upgrading.rst [hunk 5]
```diff
@@ -0,0 +1,919 @@
...
+------------------------------
+
+Pulling container images from the local Pulp server to the control plane hosts
+can take a considerable time, because images are only synced from Ark to the
+local Pulp on demand, and there is potentially a large fan-out. Pulling images
+in advance of the upgrade moves this step out of the maintenance/change window.
+
+.. code-block:: console
+
+   kayobe overcloud container image pull
```
Comment (Alex-Welsh) [inline doc/source/operations/upgrading.rst:369]: Maybe check the space on hosts before this. Something like
`kayobe overcloud host command run --command "df -h" --show-output --limit controllers[0],compute[0],storage[0]`

File: doc/source/configuration/wazuh.rst (hunk 3 above)
Comment (markgoddard) [reply doc/source/configuration/wazuh.rst:305]: It lets me do this in the upgrade doc:
```
Verify that Wazuh Manager is functioning correctly by :ref:`logging into the
Wazuh UI <wazuh-verification>`.
```

File: doc/source/operations/upgrading.rst (hunk 5 above)
Comment (markgoddard) [reply doc/source/operations/upgrading.rst:369]: Nice idea

**Comment by markgoddard** [comment]
Cancelled CI jobs to save SMS usage

**Review by Alex-Welsh — APPROVED** [review]
LGTM

**Pull request merged**